x/ccv/provider/provider_test.go

```diff
@@ -574,122 +573,122 @@ func (s *ProviderTestSuite) ReenableProviderDistribution() {
-	rspEB, _, _ := s.coordinator.CommitBlockGetResponses(cChain)
-
-	// get the packet from the endblock events
-	var packet channeltypes.Packet
-	var ftpd transfertypes.FungibleTokenPacketData
-	found = false
-	for _, evnt := range rspEB.Events {
-		if evnt.Type == channeltypes.EventTypeSendPacket {
-			found = true
-			packet, err = channelkeeper.ReconstructPacketFromEvent(evnt)
-			s.Require().NoError(err)
-			cApp.AppCodec().MustUnmarshalJSON(packet.GetData(), &ftpd)
-		}
-	}
-	s.Require().True(found)
-
-	// ensure the correct amount is being transmitted within the packet
-	expConsRedistrAmt := expFeePoolAtDistr * 3 / 4 // because of default 75% redistribute fraction (will truc decimal)
-	expProviderAmt := expFeePoolAtDistr - expConsRedistrAmt
-	s.Assert().Equal(ftpd.Amount, fmt.Sprintf("%v", expProviderAmt))
-
-	// halt provider distribution (for testing purposes to be able to review the
-	// provider fee pool)
-	s.DisableProviderDistribution()
-
-	// relay the packet
-	err = s.transferPath.RelayPacket(packet)
-	s.Require().NoError(err)
-
-	// check the consumer chain fee pool which should be now emptied (besides
-	// new minted tokens since the transfer)
-	balance = cApp.BankKeeper.GetBalance(cChain.GetContext(), consumerFeePoolAddr, "stake")
-	s.Assert().Equal(balance.Amount.Int64(), int64(26786189989304)) // this is "small"
-
-	// check the provider chain fee pool which should now have
-	// the consumer chain tokens
-	balance = pApp.BankKeeper.GetBalance(pChain.GetContext(), providerFeePoolAddr,
-		"ibc/3C3D7B3BE4ECC85A0E5B52A3AEC3B7DFC2AA9CA47C37821E57020D6807043BE9")
-	s.Assert().Equal(balance.Amount.Int64(), expProviderAmt)
-
-	// check the consumer redistribution amount arrives in the module account
-	consumerRedistrAddr := cApp.AccountKeeper.GetModuleAccount(ctx,
-		consumertypes.ConsumerRedistributeName).GetAddress()
-	balance = cApp.BankKeeper.GetBalance(cChain.GetContext(), consumerRedistrAddr, "stake")
-	s.Assert().Equal(balance.Amount.Int64(), expConsRedistrAmt)
-
-	// Ensure provider pool emptied and that allocation was called normally
-	// allocation would result in validator rewards, but due to limitations in
-	// the testing framework (where validators do not actually sign votes and
-	// therefor do not produce abci.VoteInfo) the expected behaviour of
-	// allocation is to send all rewards to the community pool
-	s.ReenableProviderDistribution()
-	s.coordinator.CommitNBlocks(pChain, 1)
-	balance = pApp.BankKeeper.GetBalance(pChain.GetContext(), providerFeePoolAddr,
-		"ibc/3C3D7B3BE4ECC85A0E5B52A3AEC3B7DFC2AA9CA47C37821E57020D6807043BE9")
-	s.Assert().Equal(balance.Amount.Int64(), int64(0))
-	communityPool := pApp.DistrKeeper.GetFeePool(pChain.GetContext()).CommunityPool
-	balanceI64 := communityPool.AmountOf(
-		"ibc/3C3D7B3BE4ECC85A0E5B52A3AEC3B7DFC2AA9CA47C37821E57020D6807043BE9").RoundInt64()
-	s.Assert().Equal(balanceI64, expProviderAmt)
-}
+// func (s *ProviderTestSuite) TestDistribution() {
```

Comment:
@mpoke
Could you ping @rigelrozanski to debug this :) ?
I can hop on call with him to help

Reply:
Yo. Okay so the distribution tests are failing _because_ mint has been removed from `app/consumer/app.go`. If there is nothing to distribute then there will be nothing to test. This test could be modified to artificially add tokens to the FeePool (basically simulating mint, or fees being spent on the consumer chain) but that doesn't seem like a great use of time, the test would need to be largely rewritten. So if the consensus is to remove mint, then I'd recommend adding a bool in the consumer app.go that _optionally_ removes mint so it can still be used for testing. OR alternatively if mint will be re-added at some later point, this test can just remain commented out for the time being.

I guess I don't really understand _why_ mint isn't a part of the MVCC? Isn't the whole idea that a consumer chain has a token which it uses for something? If it doesn't have mint, it doesn't have a token to send back... I guess just the consumer would simply send back any fees collected... which may work I suppose (thinking out loud here).

Reply:
@rigelrozanski We plan to have multiple flavors of consumer chains, each one with its own app.go. For the minimum viable consumer chain (MVCC), the mint module is disable since there is no native token. For the governance-enabled consumer chain (see https://github.com/cosmos/interchain-security/issues/141), the mint module is required. 

Regarding testing, can't we just add in the unit test some tokens to `feeCollectorName` account on the consumer chain? Does that require a lot of work?